ci/generate-traditional-boms

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+
+# prototype-kicad-project-ci-cd
+#
+# Copyright 2021, Andrew Countryman <apcountryman@gmail.com> and the
+# prototype-kicad-project-ci-cd contributors
+#
+# You may redistribute and modify this documentation and make products using it under the
+# terms of the CERN-OHL-P v2 (https:/cern.ch/cern-ohl). This documentation is distributed
+# WITHOUT ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING OF MERCHANTABILITY, SATISFACTORY
+# QUALITY AND FITNESS FOR A PARTICULAR PURPOSE. Please see the CERN-OHL-P v2 for
+# applicable conditions.
+
+# File: ci/generate-traditional-boms
+# Description: prototype-kicad-project-ci-cd repository CI traditional BOM(s) generation
+#       script.
+
+function error()
+{
+    local -r message="$1"
+
+    ( >&2 echo "$mnemonic: $message" )
+}
+
+function abort()
+{
+    if [[ "$#" -gt 0 ]]; then
+        local -r message="$1"
+
+        error "$message, aborting"
+    fi
+
+    exit 1
+}
+
+function display_help_text()
+{
+    echo "NAME"
+    echo "    $mnemonic - Generate BOM(s) for the project."
+    echo "SYNOPSIS"
+    echo "    $mnemonic --help"
+    echo "    $mnemonic --version"
+    echo "    $mnemonic"
+    echo "OPTIONS"
+    echo "    --help"
+    echo "        Display this help text."
+    echo "    --version"
+    echo "        Display the version of this script."
+    echo "EXAMPLES"
+    echo "    $mnemonic --help"
+    echo "    $mnemonic --version"
+    echo "    $mnemonic"
+}
+
+function display_version()
+{
+    echo "$mnemonic, version $version"
+}
+
+function generate_traditional_boms()
+{
+    local schematics; mapfile -t schematics < <( git -C "$repository" ls-files '*.sch' ':!:*-sheet-*.sch' | xargs -r -d '\n' -I '{}' find "$repository/{}" ); readonly schematics
+
+    for schematic in "${schematics[@]}"; do
+        if ! "$repository/utilities/KiAuto/src/eeschema_do" bom_xml "$schematic" "$repository/temp"; then
+            abort
+        fi
+    done
+
+    local boms_xml; mapfile -t boms_xml < <( find "$repository" -name '*.xml' ); readonly boms_xml
+
+    for bom_xml in "${boms_xml[@]}"; do
+        local bom
+        bom=$( basename "$bom_xml" .xml )
```

Comment:
Please place the declaration and assignment on the same line for consistency with other variables, such as the `schematics` variable.
```suggestion
        local bom; bom=$( basename "$bom_xml" .xml )
```